scripts/Muon/GUI/Common/observer_pattern.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import (absolute_import, division, print_function)
+
+
+class Observer(object):
+    """
+    The Observer observes (or subscribes) Observables, when the Observable notifies its subscribers (the observers)
+    they will cal their update() method
+    """
+
+    def __init__(self):
+        pass
+
+    def update(self, observable, arg):
+        """
+        Called when the observed object is modified. You call an Observable object's notifyObservers method
+        to notify all the object's observers of the change.
+        """
+        pass
+
+
+class Observable(object):
+    """
+    The Observable is an object which may be observed by Observers. It maintains a list of observers (aka subscribers)
+    to it, and when needed, it will notify those subscribers.
```

Comment:
Would the below work better?
```
The Observable is an object which may be subscribed to by Observers. It maintains a list of subscribers to it, and when needed, it will notify those subscribers.
```